File: frontend/src/app/services/models/categories/events/lobby/lobby-participant-event/lobby-participant-event-envelope.ts
import {WebsocketEnvelope} from '../../../../websocket-envelope';
import {LobbyClientEvent} from '../lobby-client-event/lobby-client-event';
import {LobbyParticipantEvent} from './lobby-participant-event';

export interface LobbyParticipantEventEnvelope extends WebsocketEnvelope {
  payload: LobbyParticipantEvent;
}

File: backend/src/main/java/dk/mathiaskofod/services/session/envelopes/LobbyParticipantEventEnvelope.java
package dk.mathiaskofod.services.session.envelopes;

import dk.mathiaskofod.services.session.events.lobby.participant.LobbyParticipantEvent;
import dk.mathiaskofod.services.session.models.annotations.Category;

@Category("LOBBY_PARTICIPANT_EVENT")
public record LobbyParticipantEventEnvelope(LobbyParticipantEvent payload)
        implements WebsocketEnvelope<LobbyParticipantEvent> {}
